vendor-extra/JatsContentBundle/src/EventListener/CreateView/SupListener.php

```diff
@@ -2,20 +2,19 @@
 
 declare(strict_types=1);
 
-namespace Libero\JatsContentBundle\ViewConverter;
+namespace Libero\JatsContentBundle\EventListener\CreateView;
 
 use FluentDOM\DOM\Element;
 use Libero\ViewsBundle\Views\ConvertsChildren;
-use Libero\ViewsBundle\Views\OptionalTemplateVisitor;
+use Libero\ViewsBundle\Views\OptionalTemplateListener;
 use Libero\ViewsBundle\Views\View;
 use Libero\ViewsBundle\Views\ViewConverter;
-use Libero\ViewsBundle\Views\ViewConverterVisitor;
 use function Libero\ViewsBundle\array_has_key;
 
-final class SupVisitor implements ViewConverterVisitor
```

Comment:
there's no interface anymore? It's probably inherited from the traits but worth to express it here if present?

Reply:
The equivalent is to implement [`EventSubscriberInterface`](https://github.com/symfony/symfony/blob/4.1/src/Symfony/Component/EventDispatcher/EventSubscriberInterface.php) rather than declaring it the service definition https://github.com/libero/browser/blob/e7e06056df2403657dd404c99684e3365f7fe19e/vendor-extra/JatsContentBundle/src/Resources/config/services.xml#L118 (which means it's not coupled to the Symfony dispatcher)

Reply:
Could have

```php
interface CreateViewListener
{
    public function onCreateView(CreateViewEvent $event) : void;
}
```

but it wouldn't be required.